lightningd/jsonrpc.c

```diff
@@ -741,12 +741,11 @@ bool jsonrpc_command_add(struct jsonrpc *rpc, struct json_command *command)
 
 void jsonrpc_command_remove(struct jsonrpc *rpc, const char *method)
 {
-	// FIXME: Currently leaves NULL entries in the table, if we
-	// restart plugins we should shift them out.
 	for (size_t i=0; i<tal_count(rpc->commands); i++) {
 		struct json_command *cmd = rpc->commands[i];
 		if (cmd && streq(cmd->name, method)) {
-			rpc->commands[i] = tal_free(cmd);
+			tal_arr_remove(&rpc->commands, i);
+			tal_free(cmd);
```

Comment:
Pretty sure we should break or return here?  If we really want to get multiples, you'll need to `i--` here to avoid skipping the next one...

Reply:
Also, I think we can now remove the `if (cmd &&` and similar in find_cmd and jsonrpc_command_add?